extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -483,12 +483,15 @@ export class AmpA4A extends AMP.BaseElement {
     }
     // If in localDev `type=fake` Ad specifies `force3p`, it will be forced
     // to go via 3p.
-    if (getMode().localDev &&
-        this.element.getAttribute('type') == 'fake' &&
-        this.element.getAttribute('force3p') == 'true') {
-      this.adUrl_ = this.getAdUrl();
-      this.adPromise_ = Promise.resolve();
-      return;
+    if (getMode().localDev && this.element.getAttribute('type') == 'fake') {
+      // Prevent issuing a second network request to try to fetch
+      // the same JSON again.
+      this.experimentalNonAmpCreativeRenderMethod_ = XORIGIN_MODE.NAMEFRAME;
```

Comment:
of course it's not in prod.

my point is: why you have to have this logic in test to make it work? and why prod can work without it? how does prod prevent hitting the json cache?

overall, we want to avoid introducing test-only code in non-test. does it make sense?